app.py

```diff
@@ -44,4 +44,176 @@ def show_breakdown_dialog(self):
+                new_sg_data["type"] = i["sg_data"]["type"]
+                new_sg_data["code"] = i["sg_data"]["code"]
+                new_sg_data["entity"] = i["sg_data"]["entity"]
+                new_sg_data["task"] = i["sg_data"]["task"]
+                new_sg_data["published_file_type"] = i["sg_data"]["published_file_type"]
+                new_sg_data["name"] = i["sg_data"]["name"]
+                new_sg_data["project"] = i["sg_data"]["project"]
+                new_sg_data["version_number"] = i["sg_data"]["version_number"]
+                i["sg_data"] = new_sg_data
+                
+        return items
+
+
+    def calculate_latest_version(self, template, fields):
+        """
+        Given a template and some fields, return the highest version number found on disk.
+        The template key containing the version number is assumed to be named {version}.
+        
+        This will perform a scan on disk to determine the highest version.
         
+        You typically use this in conjunction with the analyze_scene() method. 
+        For example:
+        
+        import sgtk
+        e = sgtk.platform.current_engine()
+        breakdown_app = e.apps["tk-multi-breakdown"]
+        
+        # get list of breakdown items
+        items = breakdown_app.analyze_scene()
+        
+        # now loop over all items
+        for i in items:
+        
+            # get the latest version on disk
+            latest_version = breakdown_app.calculate_latest_version(i["template"], i["fields"])
+            
+            # if our current version is out of date, update it!
+            current_version = i["fields"]["version"]
+            if latest_version > current_version:
+                
+                # make a fields dictionary representing the latest version
+                latest_fields = copy.copy(i["fields"])
+                latest_fields["version"] = latest_version
+                
+                # request that the breakdown updates to the latest version
+                breakdown_app.update_item(i["node_type"], i["node_name"], i["template"], latest_fields)
+         
+        :param template: Template object to calculate for
+        :param curr_fields: A complete set of fields for the template
+        :returns: The highest version number found
+        """
+        tk_multi_breakdown = self.import_module("tk_multi_breakdown")
+        return tk_multi_breakdown.calculate_latest_version(template, fields)
+        
+        
+    def update_item(self, node_type, node_name, template, fields):
+        """
+        Request that the breakdown updates an given node with a new version.
+        This is similar to running the update in the breakdown UI. The actual 
+        update call will be dispatched to a hook which handles the DCC specific logic.
+        
+        You typically use this in conjunction with the analyze_scene() method. 
+        For example:
+        
+        import sgtk
```

Comment:
It seems really verbose having the same full example in the doc-string of both of these methods?  Maybe instead the example could move to the doc-string for the app/class and the examples in these methods could be the specific piece that uses the method?